### PR TITLE
ENH: Refactored fio based on the recent iperf3 refactor

### LINKED ISSUE #3: Create schema definition for disk benchmark
A schema definition should be created which describes a disk benchmark.

### PR DESCRIPTION
**Pull request by julio-santiesteban**
- Added image support
 - Added command line support
 - Refactored package structure
 - Added printable status
 - Simplified kubernetes argument passing

Closes #3

### CODE REVIEW

File: controllers/fio/job.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/firepear/qsplit"
+	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
+)
+
+// NewJob creates a fio benchmark job
+func NewJob(cr *perfv1alpha1.Fio) *batchv1.Job {
```
Comment (otto-von-bivouac) [inline controllers/fio/job.go]: Can you please create a quick test for this function?

File: api/v1alpha1/fio_types.go [hunk 2]
```diff
@@ -22,23 +22,32 @@ import (
...
-	// RemoteJobFiles lists the URLs of optional remote job files. All the given
-	// files will be downloaded and you can use the `jobFiles` field to select which
-	// ones to run
-	RemoteJobFiles []string `json:"remoteJobFiles,omitempty"`
+	// Image defines the fio docker image used for the benchmark
+	Image ImageSpec `json:"image"`
+
+	// BuiltinJobFiles contains a list of fio job files that are already present
+	// in the docker image
+	BuiltinJobFiles []string `json:"builtinJobFiles,omitempty"`
```
Comment (otto-von-bivouac) [inline api/v1alpha1/fio_types.go]: +optional is missing

File: api/v1alpha1/fio_types.go [hunk 3]
```diff
@@ -22,23 +22,32 @@ import (
 
 // FioSpec defines the desired state of Fio
 type FioSpec struct {
```
Comment (otto-von-bivouac) [inline api/v1alpha1/fio_types.go]: can we please add the Affinity related structs?

Comment (julio-santiesteban) [reply api/v1alpha1/fio_types.go]: sure, but can we please do it in a separate PR, this is getting big

Comment (otto-von-bivouac) [reply api/v1alpha1/fio_types.go]: sure. can you add a TODO line for that? the bot will take care of issue creation then.

File: controllers/fio/job.go (hunk 1 above)
Comment (julio-santiesteban) [reply controllers/fio/job.go]: created

File: api/v1alpha1/fio_types.go (hunk 2 above)
Comment (julio-santiesteban) [reply api/v1alpha1/fio_types.go]: added

File: controllers/fio/suite_test.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package fio
```
Comment (otto-von-bivouac) [inline controllers/fio/suite_test.go]: Apache header is missing

**Comment by otto-von-bivouac** [comment]
LGTM. I'll quickly create a pr for the missing header in `controllers/fio/suite_test.go`.

**Pull request merged**